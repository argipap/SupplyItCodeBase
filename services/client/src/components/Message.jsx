import React from 'react';

const Message = (props) => {
    return (
        <div className={`notification is-${props.messageType}`}>
            <button className="delete" onClick={()=>{props.removeMessage()}}></button>
            <span>{props.messageName}</span>
        </div>
    )
};

export default Message;