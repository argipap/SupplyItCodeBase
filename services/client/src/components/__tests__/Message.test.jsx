import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Message from '../Message';


describe('When given a success message', () => {
    const removeMessage = jest.fn();

    const messageSuccessProps = {
        messageName: 'Hello, World!',
        messageType: 'success',
        removeMessage: removeMessage,
    }

    it(`Message renders properly`, () => {
        const wrapper = shallow(<Message {...messageSuccessProps} />);
        const element = wrapper.find('.notification.is-success');
        expect(element.length).toBe(1);
        const span = wrapper.find('span');
        expect(span.length).toBe(1);
        expect(span.get(0).props.children).toContain(
            messageSuccessProps.messageName);
        const button = wrapper.find('button');
        expect(button.length).toBe(1);
        expect(removeMessage).toHaveBeenCalledTimes(0);
        button.simulate('click');
        expect(removeMessage).toHaveBeenCalledTimes(1);
    });

    test('Message renders a snapshot properly', () => {
        const tree = renderer.create(
            <Message {...messageSuccessProps} />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe('When given a danger message', () => {
    const removeMessage = jest.fn();

    const messageDangerProps = {
        messageName: 'Hello, World!',
        messageType: 'danger',
        removeMessage: removeMessage,
    }

    it(`Message renders properly`, () => {
        const wrapper = shallow(<Message {...messageDangerProps} />);
        const element = wrapper.find('.notification.is-danger');
        expect(element.length).toBe(1);
        const span = wrapper.find('span');
        expect(span.length).toBe(1);
        expect(span.get(0).props.children).toContain(
            messageDangerProps.messageName);
        const button = wrapper.find('button');
        expect(button.length).toBe(1);
        expect(removeMessage).toHaveBeenCalledTimes(0);
        button.simulate('click');
        expect(removeMessage).toHaveBeenCalledTimes(1);
    });

    test('Message renders a snapshot properly', () => {
        const tree = renderer.create(
            <Message {...messageDangerProps} />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});