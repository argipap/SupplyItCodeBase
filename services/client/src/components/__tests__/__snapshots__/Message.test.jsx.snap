// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When given a danger message Message renders a snapshot properly 1`] = `
<div
  className="notification is-danger"
>
  <button
    className="delete"
    onClick={[Function]}
  />
  <span>
    Hello, World!
  </span>
</div>
`;

exports[`When given a success message Message renders a snapshot properly 1`] = `
<div
  className="notification is-success"
>
  <button
    className="delete"
    onClick={[Function]}
  />
  <span>
    Hello, World!
  </span>
</div>
`;
